src/NUnitFramework/framework/Constraints/ItemsConstraintExpression.cs

```diff
@@ -0,0 +1,52 @@
+// ***********************************************************************
+// Copyright (c) 2016 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// An extension of ResolvableConstraintExpression that adds a no-op Items property for readability.
+    /// </summary>
+    public sealed class ItemsConstraintExpression : ResolvableConstraintExpression
+    {
+        /// <summary>
+        /// Create a new instance of ItemsConstraintExpression
+        /// </summary>
+        public ItemsConstraintExpression() { }
+
+        /// <summary>
+        /// Create a new instance of ResolvableConstraintExpression,
+        /// passing in a pre-populated ConstraintBuilder.
+        /// </summary>
+        /// <param name="builder"></param>
+        public ItemsConstraintExpression(ConstraintBuilder builder)
+            : base(builder) { }
+
+        /// <summary>
+        /// No-op property for readability.
+        /// </summary>
+        public ResolvableConstraintExpression Items
```

Comment:
That's correct. Intentionally, `ItemsConstraintExpression.Items` returns a `ResolvableConstraintExpression` rather than an `ItemsConstraintExpression`.